Models/EventTypesListSortFilterProxyModel.cpp

```diff
@@ -0,0 +1,24 @@
+#include "EventTypesListSortFilterProxyModel.h"
+#include "EventTypesListModel.h"
+
+
+EventTypesListSortFilterProxyModel::EventTypesListSortFilterProxyModel(QObject* parent)
+    : QSortFilterProxyModel(parent) {}
+
+bool EventTypesListSortFilterProxyModel::filterAcceptsRow(int source_row, const QModelIndex& /*source_parent*/) const {
+  return sourceModel()->data(sourceModel()->index(source_row, 0), Qt::UserRole).toBool();
+}
+
+bool EventTypesListSortFilterProxyModel::lessThan(const QModelIndex& source_left,
+                                                  const QModelIndex& source_right) const {
+  //Try to sort alpha but keep grouped events together
+  QString leftGroup = sourceModel()->data(source_left, EventTypesListModel::UserRoles::EventGroupRole).toString();
+  QString rightGroup = sourceModel()->data(source_right, EventTypesListModel::UserRoles::EventGroupRole).toString();
+
+  if (leftGroup == rightGroup) {
+    return sourceModel()->data(source_left, Qt::DisplayRole).toString() <
+           sourceModel()->data(source_right, Qt::DisplayRole).toString();
+  }
+
+  return leftGroup < rightGroup;
```

Comment:
This is an alphanumeric sort, GM and LGM used sorted by the main id, which is why you have alarms and other events appearing before the create event.
![Alphanumeric Event Sort Alarms Before Create](https://user-images.githubusercontent.com/3212801/91745798-8dc89080-eb89-11ea-8e8f-0164576476dd.png)

Reply:
We do not store integer ids for events anymore so if you want a specific ordering you will need to hard code it in the sort

Reply:
It's fine, I just wanted to document it, I don't really care. The only thing I might say to do is we should just always put create first, that's the only part of the order that's stuck in my head, that the create is at the top, like a constructor usually is in a class.